test/common/upstream/load_balancer_benchmark.cc

```diff
@@ -0,0 +1,137 @@
+#include "common/runtime/runtime_impl.h"
+#include "common/upstream/ring_hash_lb.h"
+#include "common/upstream/upstream_impl.h"
+
+#include "test/common/upstream/utility.h"
+#include "test/mocks/upstream/mocks.h"
+
+#include "testing/base/public/benchmark.h"
+
+namespace Envoy {
+namespace Upstream {
+
+class RingHashTester {
+public:
+  RingHashTester(uint64_t num_hosts, uint64_t min_ring_size) {
+    HostSet& host_set = priority_set_.getOrCreateHostSet(0);
+
+    HostVector hosts;
+    for (uint64_t i = 0; i < num_hosts; i++) {
+      hosts.push_back(makeTestHost(info_, fmt::format("tcp://10.0.{}.{}:6379", i / 256, i % 256)));
+    }
+    HostVectorConstSharedPtr updated_hosts{new HostVector(hosts)};
+    host_set.updateHosts(updated_hosts, updated_hosts, nullptr, nullptr, hosts, {});
```

Comment:
wdyt of commenting the formal names for the nullptrs and {} ?  I don't see this a lot in the codebase but it sometimes helps readability without having to refer back to the .h.

come to think of it, since updated_hosts is passed twice, indicating the formal for those actuals would be nice too.

up to you, obviously

Reply:
In general I feel that with a decent editor it doesn't add that much value, but happy to do it if preferable.

Reply:
Good point.  I'll try to confer with others locally about how to get some kind of TAGS thing set up for Envoy.  But it's not just the editor; it's also the code review tool, which ideally should have hot-links for all referenced types to their declaration, but github doesn't.  Or is there a way to get github to light up some links in the review tool?

Reply:
Unfortunately I don't think GH does this.

Reply:
OK, np.  It's not a big deal to poke around and find the formals when I need to know them; just an idea.